packages/components/src/public/QueryModel/CustomizeGridViewModal.test.tsx

```diff
@@ -258,5 +258,20 @@ describe('includedColumnsForCustomizationFilter', () => {
 
         col = new QueryColumn({ name: 'testColumn', fieldKeyPath: 'Ancestors' });
         expect(includedColumnsForCustomizationFilter(col, false)).toBeTruthy();
+
+        // Issue 52337. removeFromViewCustomization is set via queryMetadata, but is meant to apply to fields only, not types
+        col = new QueryColumn({
+            name: 'Protocol',
+            fieldKeyPath: 'Ancestors/RegistryAndSources/Protocol',
+            removeFromViewCustomization: true,
+        });
+        expect(includedColumnsForCustomizationFilter(col, false)).toBeTruthy();
+
+        col = new QueryColumn({
+            name: 'Protocol',
+            fieldKeyPath: 'Ancestors/Samples/SampleType/Protocol',
+            removeFromViewCustomization: true,
+        });
+        expect(includedColumnsForCustomizationFilter(col, false)).toBeTruthy();
```

Comment:
I added a test case for the `fieldKeyPath: 'Protocol'` and `removeFromViewCustomization: true` here and expected it to be false but it wasn't. Looks like the test case above around line 235 sets the module context to have the premium module so any further test with `removeFromViewCustomization: true` will return true because of the premium check. We'll need to either reset the LABKEY.moduleContext after that test case above or set it before these cases.

Reply:
https://github.com/LabKey/labkey-ui-components/pull/1724/commits/0c7375519e16b201538744f76fe69d0f4cd36301